src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,251 @@
+        assertNull(retrieved);
+
+        ______TS("failure: null parameter");
+        assertThrows(AssertionError.class, () -> notificationsDb.getNotification(null));
+
+        // delete created account
+        notificationsDb.deleteNotification(n.getNotificationId());
+    }
+
+    @Test
+    public void testCreateNotification() throws Exception {
+        ______TS("typical success case");
+        String uuid = UUID.randomUUID().toString();
+        NotificationAttributes n = NotificationAttributes.builder(uuid)
+                .withTitle("title")
+                .withMessage("demo message")
+                .withType(NotificationType.DEPRECATION)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withStartTime(Instant.now())
+                .withEndTime(Instant.now().plusSeconds(3600))
+                .build();
+
+        notificationsDb.createEntity(n);
+
+        ______TS("duplicate notification, creation fail");
+        NotificationAttributes dup = NotificationAttributes.builder(uuid)
+                .withTitle("title2")
+                .withMessage("demo message2")
+                .withType(NotificationType.MAINTENANCE)
+                .withTargetUser(NotificationTargetUser.STUDENT)
+                .withStartTime(Instant.now())
+                .withEndTime(Instant.now().plusSeconds(3600))
+                .build();
+
+        assertThrows(EntityAlreadyExistsException.class, () -> notificationsDb.createEntity(dup));
+
+        ______TS("failure: null parameters");
+        assertThrows(AssertionError.class, () -> notificationsDb.createEntity(null));
+    }
+
+    // TODO: for extension, some fields are not allowed to be updated after shown is true
+    @Test
+    public void testUpdateNotification() throws Exception {
+        final String TITLE = "title";
+        final String TITLE_DIFF = "title_diff";
+        final String MESSAGE = "message";
+        final String MESSAGE_DIFF = "message_diff";
+        final NotificationType TYPE = NotificationType.DEPRECATION;
+        final NotificationType TYPE_DIFF = NotificationType.MAINTENANCE;
+        final NotificationTargetUser TARGET_USER = NotificationTargetUser.STUDENT;
+        final NotificationTargetUser TARGET_USER_DIFF = NotificationTargetUser.GENERAL;
+        final Instant START_TIME = Instant.now();
+        final Instant START_TIME_DIFF = START_TIME.plusSeconds(600);
+        final Instant END_TIME = START_TIME.plusSeconds(3600);
+        final Instant END_TIME_DIFF = END_TIME.plusSeconds(600);
+
+        NotificationAttributes n = notificationsDb.createEntity(NotificationAttributes.builder(UUID.randomUUID().toString())
+                .withTitle(TITLE)
+                .withMessage(MESSAGE)
+                .withType(TYPE)
+                .withTargetUser(TARGET_USER)
+                .withStartTime(START_TIME)
+                .withEndTime(END_TIME)
+                .build());
```

Comment:
Can we move all these into a separate method like `getNewNotification()`?

Reply:
Thank you for the comment. I will try to combine this with `getNewNotificationAttributes()`.